### ISSUE TITLE
HTTP/2 support

### ISSUE THREAD

**Issue by panesofglass** [owin/owin#49]
Do we need to make any adjustments for HTTP/2?

**Comment by Tratcher** [comment]
Probably, there are some significant changes at the protocol layer to account for:
1. Things like path, method, status code, etc., are now headers, how do the existing keys map?
2. Reason Phrase is obsolete.
3. Http Version is no longer a protocol field, how do the request and response owin keys change?
4. New key & Func for push promise?
5. The upgrade from HTTP/1.1 (like websockets)

Implementations I've seen so far have striven for backwards compat by mapping the new headers to the old fields so the app can't tell the difference. A similar approach may be needed here unless you want to fork owin for HTTP/2.

**Comment by serialseb** [comment]
i agree with @Tratcher for backward compat and mapping, the app should "just work" whatever version, and an http/2 server should know what to do with that. I would like to see some proposals for upgrade paths and push promises, although these could be optional modules that we could push as separate specs / keys. It all depends on market penetration I guess.

**Comment by pedroteixeira** [comment]
Hi, is there support for HTTP/2 when using .net framework 4.x? 
even if it only in Windows 10?
thanks

**Comment by Tratcher** [comment]
@pedroteixeira it actually just works on most servers. The servers used the same data structures for HTTP/1.1 and HTTP/2 for compat reasons so no adaptations were needed at the OWIN layer. If you have more specific questions about an implementation like Microsoft.Owin you can ask over at (https://github.com/aspnet/aspnetkatana).
